misc/openapi-ballerina/modules/openapi-validator/src/main/java/org/ballerinalang/openapi/validator/OpenApiValidatorUtil.java

```diff
@@ -0,0 +1,194 @@
+                    String fileName = filePath.toAbsolutePath().getFileName().toString();
+
+                    // Compile ballerina file.
+                    BLangPackage bLangPackage = compileFile(programDir, fileName);
+
+                    // If there are compilation errors do not continue the process.
+                    if (bLangPackage.diagCollector.hasErrors()) {
+                        return;
+                    }
+
+                } else {
+                    moduleName = argList.get(0);
+
+                    BLangPackage bLangPackage = OpenApiValidatorUtil
+                            .compileModule(sourceRootPath, getModuleName(moduleName));
+
+                    // If there are no compilation errors do not continue the process.
+                    if (bLangPackage.diagCollector.hasErrors()) {
+                        return;
+                    }
+                }
+            }
+        } catch (IOException | NullPointerException e) {
+//            throw LauncherUtils.createLauncherException(Messages.getException());
+        }
+    }
+
+    /**
+     * Check whether the given file is a ballerina file.
+     *
+     * @param fileName file name to be check whether a ballerina file
+     * @return {@link Boolean} true or false
+     */
+    private static boolean isBalFile(String fileName) {
+        return fileName.endsWith(".bal");
+    }
+
+    /**
+     * Compile only a ballerina module.
+     *
+     * @param sourceRoot source root
+     * @param moduleName name of the module to be compiled
+     * @return {@link BLangPackage} ballerina package
+     */
+    public static BLangPackage compileModule(Path sourceRoot, String moduleName) throws UnsupportedEncodingException {
+        emptyPrintStream = new EmptyPrintStream();
+        CompilerContext context = getCompilerContext(sourceRoot);
+        Compiler compiler = Compiler.getInstance(context);
+        // Set an EmptyPrintStream to hide unnecessary outputs from compiler.
+        compiler.setOutStream(emptyPrintStream);
+        return compiler.compile(moduleName);
+    }
+
+    /**
+     * Compile a ballerina file.
+     *
+     * @param sourceRoot  source root of the file
+     * @param packageName package name of the file
+     * @return {@link BLangPackage} ballerina package
+     */
+    public static BLangPackage compileFile(Path sourceRoot, String packageName) throws UnsupportedEncodingException {
+        emptyPrintStream = new EmptyPrintStream();
+        CompilerContext context = getCompilerContext(sourceRoot);
+        // Set the SourceDirectory to process this compilation as a program directory.
+        context.put(SourceDirectory.class, new FileSystemProgramDirectory(sourceRoot));
+        Compiler compiler = Compiler.getInstance(context);
+        // Set an EmptyPrintStream to hide unnecessary outputs from compiler.
+        compiler.setOutStream(emptyPrintStream);
+        return compiler.compile(packageName);
+    }
```

Comment:
Can't we merge these two methods using SourceType?

Reply:
yes we can, I removed unused functions for test